### ISSUE TITLE
Pull Request

### ISSUE THREAD

**Issue by tringler** [trydis/FIFA-Ultimate-Team-Toolkit#208]
Hello,

can I have a short explanation how to send a valid pull request, so I have a few code changes, but I don't know how to create a valid pull request.

**Comment by trydis** [comment]
The most important thing is creating a new branch in your forked repository, do not send a pull request from your master branch.

1) fork this repo
2) create a feature branch in your fork (my-awesome-feature)
3) add one or more commits to that branch
4) open pull request in this repo (my-awesome-feature branch --> master branch in this repo)
5) :tada: 

Also, it's better with several focused pull requests than one large with "everything".

Read these and you should be good to go:
https://help.github.com/articles/using-pull-requests/
https://yangsu.github.io/pull-request-tutorial/

**Comment by tringler** [comment]
I assume if I had implemented 3 new features I need to rollback one-by-one and create a pull request for each of them? :-/

**Comment by trydis** [comment]
You don't need to rollback anything. They're in separate commits i assume? Create one branch for each feature and [cherry-pick](http://git-scm.com/docs/git-cherry-pick) the existing commit(s) for that particular feature. Then create pull requests using the branches you've created.

**Comment by tringler** [comment]
No, it's all edited locally without using Git at all.

**Comment by trydis** [comment]
Ok, so just fork this repo, apply the changes you've done locally to the forked repo. Then push those changes to your forked repo and open pull requests.

**Comment by tringler** [comment]
That means I can package all changes in that one pull request right?

**Comment by trydis** [comment]
- The best thing would be to create one branch per feature and then one pull request for each. 
- The next best thing would be to create one branch with the features in separate commits and one pull request. 
- The worst would be one commit with all features in one pull request, which i might not accept. It's really hard to see what has changed when it's all in one big pile of code.

**Comment by tringler** [comment]
mhh ok - I will try to seperate all the changes

**Comment by tringler** [comment]
Ok - I sent the first 2 pull requests - The 3rd one is a little big bigger, so if I understood it correctly: I can wait now until you accepted (hopefully) the both requests. After that I can update the master branch & clone it to desktop again. Then I will create a new feature branch from the updated master and push my local code this and the difference should be the last pull request - right?

**Comment by trydis** [comment]
To make it as easy as possible for yourself, i would advice you to not add your commits to your master branch, like you've currently done with the "price range" feature. You will eventually run into problems since the history of your master might not match the the history of my master.

After we've worked through these pull requests i suggest you fork this repo again (or reset your master to a0ade6616cb391360e97c8c047d2cfc43a1afe7a) and take a look at [this](https://help.github.com/categories/collaborating-on-projects-using-pull-requests/) at the "[Working with forks](https://help.github.com/articles/working-with-forks/)" section. It will make your life a lot easier :smile:

In short:
- On your master, only pull from upstream(this repo)->master. That way, you'll never run into issue in case i don't merge things exactly like you've done
- Create separate branches for each new feature
- You can create other branches, such as "my-awesome-features" where you can merge all your feature branches even before they're merge into upstream

If that makes any sense :smiley_cat:

**Comment by tringler** [comment]
Ok - Just to confirm for the next pull request:

a) Update upstream to my master
b) Creating feature branche(s)
c) Commit to feature branches
d) Merge branches into a new feature set branch if I have more than one feature (optionally)

Right? :)
